### ISSUE TITLE
[FEATURE] : migrate routes page to d3.js

### ISSUE THREAD

**Issue by vogloblinsky** [compodoc/compodoc#17]
These examples are good starting-points : 
- http://mbostock.github.io/d3/talk/20111018/tree.html
- http://charts.animateddata.co.uk/datavistree/
- https://github.com/d3/d3-hierarchy/blob/master/README.md#tree
- http://www.d3noob.org/2014/01/tree-diagrams-in-d3js_11.html

**Comment by basarat** [comment]
I've done stuff like this before e.g. using d3 to generate the AST https://basarat.gitbooks.io/alm/content/features/ast.html

In hindsight its better just done with DOM e.g. tree view https://basarat.gitbooks.io/alm/content/features/tree.html 

Might even be able to use the component from here : https://astexplorer.net/ 

Ofcourse feel free to have as much fun as you want :rose: :heart: and thanks for making this OSS

**Comment by vogloblinsky** [comment]
Thanks @basarat for your links.
The d3.js tree view seems to be the better option for me for displaying Angular routes.

Regarding all your amazing stuff with Typescript, i have a question for you.
I have implemented the source code view yesterday for compodoc: 

![source code view](https://raw.githubusercontent.com/groupe-sii/compodoc/master/screenshots/7.png)

I have tried to implement the 'GotoDeclaration' link for functions or properties of an Angular component or just a simple Typescript/ES6 class.
Typescript parser give us just 'pos' and 'end' information.
Did you have an idea for getting the line of the declaration with the 'start' information ?
I have started to code something with number of characters, etc, but didn't continue at all the stuff
Thanks.
